src/meta/api/src/schema_api_impl.rs

```diff
@@ -2314,6 +2322,172 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         Ok(CountTablesReply { count })
     }
 
+    async fn get_table_mutation_lock(
+        &self,
+        req: GetTableMutationLockReq,
+    ) -> Result<GetTableMutationLockReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let table_id = req.table_id;
+
+        let tbid = TableId { table_id };
+
+        let (tb_meta_seq, tb_meta): (_, Option<TableMeta>) = get_pb_value(self, &tbid).await?;
+
+        if tb_meta_seq == 0 {
+            return Err(KVAppError::AppError(AppError::UnknownTableId(
+                UnknownTableId::new(table_id, "get_table_mutation_lock"),
+            )));
+        }
+
+        debug!(
+            ident = display(&tbid),
+            table_meta = debug(&tb_meta),
+            "get_table_mutation_lock"
+        );
+
+        let lock_key = TableMutationLockKey { table_id };
+        let (lock_key_seq, _): (_, Option<TableMutationLock>) =
+            get_pb_value(self, &lock_key).await?;
+
+        Ok(GetTableMutationLockReply {
+            locked: lock_key_seq > 0,
+        })
+    }
+
+    async fn add_table_mutation_lock(
+        &self,
+        req: AddTableMutationLockReq,
+    ) -> Result<AddTableMutationLockReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let mut retry = 0;
+        let table_id = req.table_id;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
+
+            let tbid = TableId { table_id };
+            let (tb_meta_seq, tb_meta): (_, Option<TableMeta>) = get_pb_value(self, &tbid).await?;
+            if tb_meta_seq == 0 {
+                return Err(KVAppError::AppError(AppError::UnknownTableId(
+                    UnknownTableId::new(table_id, "add_table_mutation_lock"),
+                )));
+            }
+
+            debug!(
+                ident = display(&tbid),
+                table_meta = debug(&tb_meta),
+                "add_table_mutation_lock"
+            );
+
+            let lock_key = TableMutationLockKey { table_id };
+
+            let lock = TableMutationLock {};
+
+            let condition = vec![
+                txn_cond_seq(&tbid, Eq, tb_meta_seq),
```

Comment:
do we need to check the table mete sequence (version) in KV transaction? seems lock by table id is enough.  
the `retry` logic also seems to be unnecessary.